ha/core/controllers/pcs/node_controller.py

```diff
@@ -152,6 +149,119 @@ def status(self, nodeids: list = None) -> dict:
         """
         raise HAUnimplemented("This operation is not implemented.")
 
+    def _validate_permissions(self) -> None:
+
+        # confirm that user is root or part of haclient group"
+        user = getpass.getuser()
+        group_id = os.getgid()
+
+        try:
+            # find group id for root and haclient
+            id_ha = grp.getgrnam(const.USER_GROUP_HACLIENT)
+            id_root = grp.getgrnam(const.USER_GROUP_ROOT)
+
+            # if group not root or haclient return error
+            if group_id != id_ha.gr_gid and group_id != id_root.gr_gid:
+                Log.error(f"User {user} does not have necessary permissions to execute this CLI")
+                raise HAInvalidPermission(
+                            f"User {user} does not have necessary permissions to execute this CLI")
+
+            # The user name "hauser"; which is part of the "haclient" group;
+            # is used by HA.
+            # internal commands are allowed only if the user is "hauser"
+            # As of now, every HA CLI will be internal command. So, we
+            # do not need this change. We can revisit this if needed in future
+            #if user == const.USER_HA_INTERNAL:
+            #    self._is_hauser = True
+
+
+        # TBD : If required raise seperate exception  for root and haclient
```

Comment:
Fix the TBD

Reply:
Exception is not required for root user which is by default user. removed TBD

Reply:
TBD is still present. Remove it